gen/generate.go

```diff
@@ -51,7 +51,7 @@ type genesisAllocation struct {
 }
 
 // GenerateGenesisFiles generates the genesis.json file and wallet files for a give genesis configuration.
-func GenerateGenesisFiles(genesisData GenesisData, outDir string, verbose bool) error {
+func GenerateGenesisFiles(genesisData GenesisData, outDir string, verbose bool, consensus map[protocol.ConsensusVersion]config.ConsensusParams) error {
```

Comment:
consensus is in addition to config.Consensus. It will be nice if the name reflects that. 
Suggestions: optionalConsensus or testConsensus

Reply:
and another nit: a new param is kind of "src" param, and outDir is "dst". So it might make sense to put the new param as the second arg of the function.

Reply:
or better, configurableConsensus as is used elsewhere.

Reply:
I've changes the function signature so it will receive a merged consensus as an input.